### PR TITLE
[Docs] Add caching topics to `ROADMAP.md`

### PR DESCRIPTION
**Pull request by foundrytom**

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ host.
 
 ## Y24 onwards
 
+- Change tracking and out-the-box caching mechanisms.
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply ROADMAP.md:55]: 🤔 Was trying to avoid a specific solution, was meant to be "Entity change tracking"

Comment (elliotcmorris) [inline ROADMAP.md:55]:
```suggestion
- Change-tracking and out-the-box caching mechanisms.
```

[Consider] Sorry to leave a comment on such a small PR, but it actually took me like a minute to figure out this was about tracking changes, not changing some mysterious `tracking` feature.

**Review by elliotcmorris — APPROVED** [review]

File: ROADMAP.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (feltech) [reply ROADMAP.md:55]: Prefixing with "entity" makes it a bit more obvious.

**Comment by feltech** [comment]
> @elliotcmorris @feltech I took the liberty of force-pushing.

But now I can't tell what changed :grin:

**Review by feltech — APPROVED** [review]

**Comment by foundrytom** [comment]
> > @elliotcmorris @feltech I took the liberty of force-pushing.
> 
> But now I can't tell what changed 😁

Maybe we should move to GitLab

File: ROADMAP.md (hunk 1 above)
Comment (elliotcmorris) [reply ROADMAP.md]: :+1:

**Comment by elliotcmorris** [comment]
Sorry for the delay, now you have to do a rebase. :(
RAMWR

**Review by elliotcmorris — APPROVED** [review]

**Pull request merged**